resizer.go

```diff
@@ -453,7 +453,13 @@ func imageCalculations(o *Options, inWidth, inHeight int) float64 {
 	switch {
 	// Fixed width and height
 	case o.Width > 0 && o.Height > 0:
-		factor = math.Min(xfactor, yfactor)
+		// Use minimum factor for croping or downscaling
+		// use maximum factor for upscaling
+		if o.Crop || (o.Width < inWidth || o.Height < inHeight) {
+			factor = math.Min(xfactor, yfactor)
+		} else {
+			factor = math.Max(xfactor, yfactor)
+		}
```

Comment:
fyi. this is a revert from #198 with an additional check. Might be better, indeed.

Reply:
Thanks for the hint. Wat was the reason for removing the check in first place?

Reply:
tests were breaking.

Reply:
I see, `TestResizeVerticalImage` fails with #198.
We see some shrinking issues introduced with #198, currently still under investigation.

Reply:
@greut please see #205